### PR TITLE
chore: improve list-feeds command

### PR DESCRIPTION
**Pull request by Ananya2001-an**
Added the following things:
- pretty flag
- predicate option to be dependent on bbox input
- output to stdout (each url on newline) when pretty=False
- updated mypy config in pyproject....it still raises many errors so I have kept it commented inside the pre-commit config file
- updated tests accordingly

### CODE REVIEW

File: gtfs/__main__.py [hunk 1]
```diff
@@ -1,22 +1,25 @@
...
+from .utils.constants import Predicate, spinner
 from .utils.geom import Bbox, bbox_contains_bbox, bbox_intersects_bbox
 
 logging.basicConfig()
 LOG = logging.getLogger()
 app = typer.Typer()
 
 
-def check_bbox(bbox: str) -> Bbox:
+def check_bbox(bbox: str) -> Union[Bbox, None]:
```
Comment (nilsnolde) [inline gtfs/__main__.py]:
```suggestion
def check_bbox(bbox: str) -> Optional[Bbox]
```

File: gtfs/__main__.py [hunk 2]
```diff
@@ -39,50 +42,76 @@ def check_bbox(bbox: str) -> Bbox:
...
+        )
+    else:
+        spinner("Fetching feeds...", 1)
+        typer.secho("Filtered feeds are:", fg=typer.colors.BLUE)
+        if pretty is True:
+            output = ColorTable(
+                ["Feed Source", "Transit URL", "Bounding Box"], theme=Themes.OCEAN, hrules=1
+            )
+        else:
+            output = []
```
Comment (nilsnolde) [inline gtfs/__main__.py]:
```suggestion
        pretty_output = ColorTable(
                ["Feed Source", "Transit URL", "Bounding Box"], theme=Themes.OCEAN, hrules=1
```

File: gtfs/__main__.py [hunk 3]
```diff
@@ -39,50 +42,76 @@ def check_bbox(bbox: str) -> Bbox:
...
+                ]
+                output.add_row(row)
+            else:
+                output.append(src.url)
+
+        if pretty is True:
+            print("\n" + output.get_string())
+        else:
+            print("\n".join(output))
+        typer.secho("All done!", fg=typer.colors.GREEN)
```
Comment (nilsnolde) [inline gtfs/__main__.py]: I'd remove this line, should be blazingly fast, no need for this. More importantly, I'll mess up the default of printing to stdout and piping into other command(s)

File: gtfs/__main__.py [hunk 4]
```diff
@@ -39,50 +42,76 @@ def check_bbox(bbox: str) -> Bbox:
...
+                    [feed_bbox.min_x, feed_bbox.min_y, feed_bbox.max_x, feed_bbox.max_y],
+                ]
+                output.add_row(row)
+            else:
+                output.append(src.url)
+
+        if pretty is True:
+            print("\n" + output.get_string())
+        else:
+            print("\n".join(output))
```
Comment (nilsnolde) [inline gtfs/__main__.py]:
```suggestion
            if pretty is True:
                output.add_row([
                    src.__name__,
                    src.url,
                    [feed_bbox.min_x, feed_bbox.min_y, feed_bbox.max_x, feed_bbox.max_y],
                ])
                continue
            
            print(src.url)

        if pretty is True:
            print("\n" + output.get_string())
```

File: gtfs/__main__.py [hunk 5]
```diff
@@ -39,50 +42,76 @@ def check_bbox(bbox: str) -> Bbox:
...
+        raise typer.BadParameter(
+            f"Please pass a bbox if you want to filter feeds spatially based on predicate = {predicate}!"
+        )
+    elif bbox is not None and predicate is None:
+        raise typer.BadParameter(
+            f"Please pass a predicate if you want to filter feeds spatially based on bbox = {bbox}!"
+        )
+    else:
+        spinner("Fetching feeds...", 1)
+        typer.secho("Filtered feeds are:", fg=typer.colors.BLUE)
```
Comment (nilsnolde) [inline gtfs/__main__.py]: Same for this line, I think it's better to have a clean output, either just the table or just the newline separated names of the gtfs feeds.

**Review by nilsnolde — COMMENTED** [review]
Thanks!

Just a few nits to make the code a bit more concise and less branched.

**Comment by Ananya2001-an** [comment]
Oh yess the code would look much better this way... :)

**Review by nilsnolde — APPROVED** [review]

**Comment by nilsnolde** [comment]
Just checking before merging: this was done right?

**Comment by Ananya2001-an** [comment]
> Just checking before merging: this was done right?

Done right meaning?

**Comment by nilsnolde** [comment]
Done from your side, ready for final review/merge? If so, best to just re-request a review in the future, then it's not ambiguous.

**Comment by Ananya2001-an** [comment]
Ohh yes.. it's done from my side and sure I will request a review every time I am sure about the final changes :)

**Comment by Ananya2001-an** [comment]
I will continue to push changes for the fetch feeds command in the same PR... it's not ready for review yet but u can still look at the code simultaneously... :)

**Pull request merged**